### PR TITLE
Fix vue material icons

### PR DESCRIPTION
**Pull request by skjnldsv**
- [x] Move to md icons
  - [x] Replace
  - [x] Sizes and design match checks
  - [x] Regression checks
- [x] Cleanup old iconfont
- [x] Cleanup webpack config

For all that were possible, I switched to the material design icons straight away.
For the others (like pure css without slots or our custom statuses), I moved back to a simple url-loader and a `theme--dark` selector.

It will clean the already-too-complicated webpack config, as well as reducing the size of the final build. I took the liberty of fixing some blatant typos and changed a few docs.

The icons used are not 16px, because MaterialDesign ships 24px with margins.
So we're on 24px now :warning:

____
<details>
<summary>🖼️ Screenshots from this doc where everything is changed (Click to expand)</summary>

Capture d’écran_2021-04-01_17-58-21
Capture d’écran_2021-04-01_18-12-54
Capture d’écran_2021-04-01_18-13-26
Capture d’écran_2021-04-01_18-13-38
Capture d’écran_2021-04-01_18-15-34
Capture d’écran_2021-04-01_18-16-06
Capture d’écran_2021-04-01_18-17-26
Capture d’écran_2021-04-01_18-18-28
Capture d’écran_2021-04-01_18-18-57
Capture d’écran_2021-04-01_18-19-13
Capture d’écran_2021-04-01_18-19-30
Capture d’écran_2021-04-01_18-19-36
Capture d’écran_2021-04-01_18-19-44
Capture d’écran_2021-04-01_18-21-48
Capture d’écran_2021-04-01_18-22-11
</details>

### CODE REVIEW

File: src/components/ColorPicker/ColorPicker.vue [hunk 1]
```diff
@@ -204,16 +224,30 @@ export default {
...
+
 		pickColor(color) {
 			if (typeof color !== 'string') {
 				color = this.currentColor.hex
 			}
 			this.currentColor = color
+
+			/**
+			 *
+			 */
```
Comment (jotoeri) [inline src/components/ColorPicker/ColorPicker.vue:243]: Was there some text planned? 😄

File: src/components/SettingsSection/SettingsSection.vue [hunk 2]
```diff
@@ -48,7 +48,9 @@ This component is to be used in the settings section of nextcloud.
 				:href="docUrl"
 				class="settings-section__info"
 				role=""
-				:title="docTitleTranslated" />
+				:title="docTitleTranslated">
+				<Information :size="24" decorative />
```
Comment (jotoeri) [inline src/components/SettingsSection/SettingsSection.vue:52]: Hm, depending on where i hover, i get either the right text or just the icon-aria-label
grafik
grafik

**Review by jotoeri — COMMENTED** [review]
Yeah! The icons just disappeared on the bundles... AppNavigation falls from 50 KiB to 20 KiB 😅  Nice step! 🎉 
Honestly i didn't check all the css in detail, but generally looks good to me... ☺️

File: src/components/SettingsSection/SettingsSection.vue [hunk 3]
```diff
@@ -59,12 +61,14 @@ This component is to be used in the settings section of nextcloud.
 </template>
 
 <script>
+import Information from 'vue-material-design-icons/Information'
```
[the remark this answers is not in the document]
Comment (skjnldsv) [reply src/components/SettingsSection/SettingsSection.vue:64]: Good call, let's ask for @jancborchardt :mag:

File: src/components/AppNavigationToggle/AppNavigationToggle.vue [hunk 4]
```diff
@@ -27,14 +27,21 @@
 		:aria-expanded="open ? 'true' : 'false'"
 		aria-controls="app-navigation-vue"
 		@click.prevent="toggleNavigation"
-		@keydown.space.exact.prevent="toggleNavigation" />
+		@keydown.space.exact.prevent="toggleNavigation">
+		<Menu :size="20" decorative />
```
[the remark this answers is not in the document]
Comment (skjnldsv) [reply src/components/AppNavigationToggle/AppNavigationToggle.vue:31]:
```suggestion
		<Menu :size="24" decorative />
```

File: src/components/Content/Content.vue [hunk 5]
```diff
@@ -22,25 +22,40 @@
...
-	</template>
-	```
+```vue
+<template>
+	<Content app-name="forms">
+		<AppNavigation>
+				<template #list>
+			<AppNavigationNew text="Create article" />
+			<AppNavigationItem title="My title" icon="icon-category-enabled" />
+				</template>
```
[the remark this answers is not in the document]
Comment (skjnldsv) [reply src/components/Content/Content.vue:40]:
```suggestion
				<template #list>
					<AppNavigationNew text="Create article" />
					<AppNavigationItem title="My title" icon="icon-category-enabled" />
				</template>
```

File: src/components/SettingsSection/SettingsSection.vue (hunk 2 above)
Comment (skjnldsv) [reply src/components/SettingsSection/SettingsSection.vue:52]: Ah, I guess I need to set the title to empty text for all icons :grin:

File: src/components/AppNavigationItem/InputConfirmCancel.vue [hunk 6]
```diff
@@ -101,28 +115,19 @@ export default {
...
 				}
 			}
-
-			&.icon-confirm {
-				@include iconfont('confirm');
-				border-left: none;
-			}
-			&.icon-confirm:hover {
-				border-radius: 0px var(--border-radius) var(--border-radius) 0px !important;
-			}
```
[the remark this answers is not in the document]
Comment (skjnldsv) [reply src/components/AppNavigationItem/InputConfirmCancel.vue:121]: Peek 02-04-2021 10-29

Good enough @jotoeri ?

File: src/components/SettingsSection/SettingsSection.vue (hunk 2 above)
Comment (skjnldsv) [reply src/components/SettingsSection/SettingsSection.vue]: Done :rocket:

File: src/components/AppNavigationItem/InputConfirmCancel.vue [hunk 7]
```diff
@@ -28,24 +28,44 @@
...
 				type="text"
-				class="app-navigation-entry__inline-input"
+				class="app-navigation-input-confirm__input"
 				:placeholder="placeholder">
+
 			<button type="submit"
-				class="icon-confirm"
-				@click.stop.prevent="confirm" />
+				class="app-navigation-input-confirm__confirm"
+				:aria-label="t('Confirm changes')"
```
Comment (jotoeri) [inline src/components/AppNavigationItem/InputConfirmCancel.vue:43]: Ahh, sorry i didnt realise before: that might produce the same problem on translation extraction as we currently had...
Same below on the other button.

**Review by jotoeri — APPROVED** [review]
One thing - but apart of that. 👍 🎉

File: src/components/AppNavigationItem/InputConfirmCancel.vue (hunk 7 above)
Comment (skjnldsv) [reply src/components/AppNavigationItem/InputConfirmCancel.vue:43]: Ah, forgot about this! :+1:

Comment (skjnldsv) [reply src/components/AppNavigationItem/InputConfirmCancel.vue]: Done! :heart:

**Comment by raimund-schluessler** [comment]
There is (literally) a small issue with the last arrow in the breadcrumbs bar. The longer the text of the last crumb is, the smaller the arrow gets:

Screenshot_2021-04-02 Nextcloud Vue Style Guide

File: src/components/Breadcrumb/Breadcrumb.vue [hunk 8]
```diff
@@ -228,12 +226,21 @@ export default {
...
 
-	&::before {
+	&::after {
+		content: '';
 		display: flex;
 		align-items: center;
-		order: 1;
 		color: var(--color-border-dark);
 		font-size: 26px;
+		width: 8px;
```
Comment (raimund-schluessler) [inline src/components/Breadcrumb/Breadcrumb.vue:235]:
```suggestion
		width: 8px;
		min-width: 8px;
```

Comment (raimund-schluessler) [reply src/components/Breadcrumb/Breadcrumb.vue:235]: Seems to fix https://github.com/nextcloud/nextcloud-vue/pull/1816#issuecomment-812667792

**Review by JonathanTreffler — APPROVED** [review]
Reducing the bundles size is great and necessary. Seems good to me :)

**Review by raimund-schluessler — DISMISSED** [review]
See https://github.com/nextcloud/nextcloud-vue/pull/1816#issuecomment-812667792

**Comment by skjnldsv** [comment]
> There is (literally) a small issue with the last arrow in the breadcrumbs bar. The longer the text of the last crumb is, the smaller the arrow gets:
> 
> Screenshot_2021-04-02 Nextcloud Vue Style Guide
> 
> Besides of that, it seems work well and reduced the bundle size of my app by 250 kB.

Weirdly, this was not present in the documentation.
SO I assume there is a conflicting rule on server?

https://60672dfeacb72f0007de3ed6--nextcloud-vue-components.netlify.app/

**Review by skjnldsv — COMMENTED** [review]
We need to release v3.9.0 before merging

**Comment by raimund-schluessler** [comment]
> 
> 
> > There is (literally) a small issue with the last arrow in the breadcrumbs bar. The longer the text of the last crumb is, the smaller the arrow gets:
> > Screenshot_2021-04-02 Nextcloud Vue Style Guide
> > Besides of that, it seems work well and reduced the bundle size of my app by 250 kB.
> 
> @raimund-schluessler
> Weirdly, this was not present in the documentation.
> So I assume there is a conflicting rule on server?
> 
> https://60672dfeacb72f0007de3ed6--nextcloud-vue-components.netlify.app/

This screenshot is from the netlify docs. But you need to change the title of the last crumb to something longer. The default one is to short to trigger the issue.

**Review by raimund-schluessler — APPROVED** [review]

**Review by juliushaertl — APPROVED** [review]
Very nice :+1:

**Pull request merged**